api/envoy/extensions/transport_sockets/tls/cert_selectors/on_demand_secret/v3/config.proto

```diff
@@ -0,0 +1,37 @@
+syntax = "proto3";
+
+package envoy.extensions.transport_sockets.tls.cert_selectors.on_demand_secret.v3;
+
+import "envoy/config/core/v3/config_source.proto";
+import "envoy/config/core/v3/extension.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.transport_sockets.tls.cert_selectors.on_demand_secret.v3";
+option java_outer_classname = "ConfigProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/transport_sockets/tls/cert_selectors/on_demand_secret/v3;on_demand_secretv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: On-demand secret certificate selector]
+// [#extension: envoy.tls.certificate_selectors.on_demand_secret]
+
+// Fetches the secret on-demand while allowing the parent cluster or listener to accept connections
+// without warming. The certificate secret is validated against the outer common TLS context, e.g.
```

Comment:
Maybe document that the handshakes are paused until the cert can be retrieved from SDS.